src/java/org/apache/cassandra/index/sai/StorageAttachedIndexBuilder.java

```diff
@@ -206,8 +206,9 @@ private boolean indexSSTable(SSTableReader sstable, Set<StorageAttachedIndex> in
                         }
                     }
 
-                    bytesProcessed += keys.getDataPosition() - previousKeyPosition;
-                    previousKeyPosition = keys.getDataPosition();
+                    long dataPosition = keys.advance() ? keys.dataPosition() : sstable.uncompressedLength();
```

Comment:
Nit: I'd move `keys.advance()` to a separate line for readability -- the current form suggests the advance is only needed to calculate the data position, which is not true.

Reply:
ok